test/TypeTraitsTest.cpp

```diff
@@ -62,31 +72,36 @@ TEST(TypeTraits, SameAsAnyTypeIn) {
   using nested = std::tuple<tup>;
 
   // Successful comparison.
-  forEachTypeInTemplateType<tup>(
-      []<typename TupType>() { static_assert(SameAsAnyTypeIn<TupType, tup>); });
+  forEachTypeInTemplateTypeWithTI(ti<tup>, [](auto t) {
+    using TupType = decltype(t)::type;
+    static_assert(SameAsAnyTypeIn<TupType, tup>);
+  });
   static_assert(SameAsAnyTypeIn<tup, nested>);
 
   // Unsuccessful comparison, where the underlying type is wrong.
-  forEachTypeInParameterPack<tup, size_t, double, unsigned char>(
-      []<typename WrongType>() {
-        callLambdaWithAllVariationsOfType<WrongType>([]<typename T>() {
-          static_assert(!SameAsAnyTypeIn<WrongType, tup>);
-        });
+  forEachTypeInParameterPackWithTI<tup, size_t, double, unsigned char>(
+      [](auto t) {
+        using WrongType = decltype(t)::type;
+        callLambdaWithAllVariationsOfType<WrongType>(
+            [](auto) { static_assert(!SameAsAnyTypeIn<WrongType, tup>); });
       });
 
   // Unsuccessful comparison, where the underlying type is contained, but not
   // with those qualifiers.
-  auto testNotIncludedWithThoseQualifiers = []<typename TemplatedType>() {
-    forEachTypeInTemplateType<TemplatedType>([]<typename CorrectType>() {
-      callLambdaWithAllVariationsOfType<CorrectType>([]<typename T>() {
+  auto testNotIncludedWithThoseQualifiers = [](auto t) {
+    using TemplatedType = decltype(t)::type;
+    forEachTypeInTemplateTypeWithTI(ti<TemplatedType>, [](auto t) {
```

Comment:
```suggestion
    forEachTypeInTemplateTypeWithTI(t, [](auto t) {
```
(no need for unp0acking and repacking).

Reply:
Fixed